src/services/flowsheet.service.ts

```diff
@@ -66,12 +68,80 @@ export const getEntriesByRange = async (startId: number, endId: number) => {
   return response;
 };
 
+export const getEntriesByShow = async (show_id: number) => {
+  const response: IFSEntry[] = await db
+    .select({
+      id: flowsheet.id,
+      show_id: flowsheet.show_id,
+      album_id: flowsheet.album_id,
+      artist_name: flowsheet.artist_name,
+      album_title: flowsheet.album_title,
+      track_title: flowsheet.track_title,
+      record_label: flowsheet.record_label,
+      rotation_id: flowsheet.rotation_id,
+      rotation_play_freq: rotation.play_freq,
+      request_flag: flowsheet.request_flag,
+      message: flowsheet.message,
+      play_order: flowsheet.play_order,
+    })
+    .from(flowsheet)
+    .leftJoin(rotation, eq(rotation.id, flowsheet.rotation_id))
+    .where(eq(flowsheet.show_id, show_id))
+    .orderBy(desc(flowsheet.play_order));
+
+  return response;
+};
+
 export const addTrack = async (entry: NewFSEntry): Promise<FSEntry> => {
+
+  const matching_albums = await db.select()
+  .from(library_artist_view)
+  .where(and(
+    eq(library_artist_view.artist_name, entry.artist_name ?? ""),
+    eq(library_artist_view.album_title, entry.album_title ?? ""),
+    eq(library_artist_view.label, entry.record_label ?? "")
```

Comment:
Is there a runtime cost to leaving in equality checks for empty strings?

Reply:
There is. This was a bad way to do it; I've now updated drizzle to version 0.29 and used their [dynamic query builder interface](https://orm.drizzle.team/docs/dynamic-query-building) to solve this problem. Will probably want to consult as to whether this upgrade is desired; while I'm willing to make the proper adjustments (they're small) it does mean we're using some deprecated, though not non-functional, interfaces.